database_cleanup/model/purge_modules.py

```diff
@@ -54,7 +55,11 @@ def _module_data_uninstall(self, cr, uid, modules_to_remove, context=None):
                 ctx[MODULE_UNINSTALL_FLAG] = True
                 field = self.pool[this.model].browse(
                     cr, uid, this.res_id, context=ctx)
-                if not self.pool.get(field.model):
```

Comment:
Thanks for the bug report. In the case that you describe that the fields does not exist in `ir.model.fields` anymore, shouldn't then `field` already evaluate to `False` and the condition `if not field or not self.pool.get(field.model):` suffice to avoid the error? I think the `try`/`catch` is a bit too much.